### DISCUSSION TITLE
Lesson 13: Unit test_stake_token query

### DISCUSSION THREAD

**Discussion by wisemoon88** [smartcontractkit/full-blockchain-solidity-course-py#1342] [Q&A] [answered]
Hi everyone,

loving the course, and almost at the homestretch :)

i dont have an error, but just had a query, when we run the test_stake_token unit test, since this is a local test environment where the accounts are spun up when the test are being run, how is there already dapp tokens that exist in the account that can be staked?  maybe there's something i'm not seeing, do we already allocate a certain amount of Dapptokens to the account?

```def test_stake_tokens(amount_staked):
    # Arrange
    if network.show_active() not in LOCAL_BLOCKCHAIN_ENVIRONMENTS:
        pytest.skip("Only for local testing!")
    account = get_account()
    token_farm, dapp_token = deploy_token_farm_and_dapp_token()
    # Act
    dapp_token.approve(
        token_farm.address, amount_staked, {"from": account}
    )  # this is approving the token farm to utilize the amount staked from our accounts
    token_farm.stakeTokens(
        amount_staked, dapp_token.address, {"from": account}
    )  # this is staking the amount from our account
    # Assert
    assert (
        token_farm.stakingBalance(dapp_token.address, account.address)
        == amount_staked  # this is checking the stakers balance is equal to the amount staked
    )
    assert (
        token_farm.uniqueTokensStaked(account.address) == 1
    )  # this is checking that only one unique instance of the token is available currently
    assert (
        token_farm.stakers(0) == account.address
    )  # this is checking that the staker at 0 index is currently the same as the account that called the stake token function
    return token_farm, dapp_token```

**Comment by wisemoon88** [comment, marked as answer]
owh, i think i already figured it out.  when we deploy the dapptoken contract, the total amount of dapptoken minted is minted to the account that deploys the contract. Since we do not transfer the whole supply of tokens to tokenfarm, there is still an available amount of tokens in our account to be staked

i think that is the answer but feel free to correct me if i am wrong, 

Cheers!!

**Reply by cromewar** [reply]
Hello there @wisemoon88 you are right and you figured out by yourself, sorry for the delay on answering.
Feel free to make any other questions.

**Reply by wisemoon88** [reply]
no problem @cromewar!  thank you very much for the affirmation :)

appreciate your time and in replying to my query.

will definitely start to be more involved with the discussion community.  the people are awesome!!
